backend/app/app/crud/crud_metrics.py

```
@@ -1,82 +1,129 @@
-    ) -> Metrics:
+        self, db: Session, metrics_in: MetricGet, platform_id: int, interval: int
+    ) -> Metric:
         # TODO: implement this function
-        metrics = Metrics(**metrics_in.dict())
+        metrics = Metric(**metrics_in.dict())
         metrics.resurrection = -1
         return metrics
 
+    def get_analysis(
+        self, db: Session, metrics_in: MetricGet,
+    ) -> MetricAnalysis:
+        metric_this_interval = metrics_in[len(metrics_in)-1]
+        metric_last_interval = metrics_in[len(metrics_in)-2]
+        growth_positive = False
+        retention_positive = False
+        analysis = None
+        if metric_this_interval.growth > metric_last_interval.growth:
+            growth_positive = True
+        if metric_this_interval.retention > metric_last_interval.retention:
+            retention_positive = True
+        if growth_positive and retention_positive:
+            analysis = MetricsAnalysisEnum.permanent_growth
+        elif growth_positive and not retention_positive:
+            analysis = MetricsAnalysisEnum.leaking_bucket
+        elif not growth_positive and retention_positive:
+            analysis = MetricsAnalysisEnum.old_age
+        elif not growth_positive and not retention_positive:
+            analysis = MetricsAnalysisEnum.dead_spiral
+        return analysis
+
     def get_metrics(
-        self, db: Session, metrics_in: MetricsGet,
-    ) -> Metrics:
-        metrics = self.get_growth(db=db, metrics_in=metrics_in)
-        metrics = self.get_retention(db=db, metrics_in=metrics)
-        metrics = self.get_resurrection(db=db, metrics_in=metrics)
-        return metrics
+        self, db: Session, metrics_in: MetricGet,
+    ) -> MetricsResult:
+        metrics = []
+        metrics_list = self.generate_empty_metrics_by_interval(db=db, metrics_in=metrics_in)
+        metricsResult = MetricsResult(
+            **metrics_in.dict(), metrics_list=[], analysis=MetricsAnalysisEnum.permanent_growth)
```

Setting it to permanent_growth as default is misleading